storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -669,4 +689,10 @@ public static String remoteTransactionIndexFileName(RemoteLogSegmentMetadata rem
         return generateFileNamePrefixForIndex(remoteLogSegmentMetadata) + LogFileUtils.TXN_INDEX_FILE_SUFFIX;
     }
 
+    public static int estimatedEntrySize(Entry entry) {
+        return entry.offsetIndex.sizeInBytes() + (int) entry.offsetIndex.length() +
```

Comment:
"Entry" is immutable so the size won't change over the lifetime of this instance. Can we calculate the size once when we fetch it (in the constructor) and then store it in a member variable? This will prevent calculating sizes every time eviction algorithm runs.